src/mbgl/storage/request_holder.hpp

```diff
@@ -0,0 +1,26 @@
+#ifndef MBGL_STORAGE_REQUEST_HOLDER
+#define MBGL_STORAGE_REQUEST_HOLDER
+
+#include <memory>
+
+namespace mbgl {
+
+class Request;
+
+class RequestHolder {
```

Comment:
What's the reason for a separate `RequestHolder` class, rather than the usual pattern of returning a `std::unique_ptr<Request>` and having `~Request` auto-cancel?